### ISSUE TITLE
[android] Device test failing in CustomGeometrySource

### ISSUE THREAD

**Issue by friedbunny** [mapbox/mapbox-gl-native#12618]
A changelog-only PR (#12462) branched off of current master (de59e3a9ce0d8371f584b31f5312cb20d717a96f) is repeatedly [failing](https://circleci.com/gh/mapbox/mapbox-gl-native/142205) `android-debug-arm-v7` [with](https://console.firebase.google.com/project/android-gl-native/testlab/histories/bh.1470d2ab45df6f1f/matrices/4852805886422376216/executions/bs.262fea2ac80bc7e7):

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'google/sailfish/sailfish:8.0.0/OPR3.170623.013/4397526:user/release-keys'
Revision: '0'
ABI: 'arm'
pid: 9844, tid: 10035, name: Thread-2596  >>> com.mapbox.mapboxsdk.testapp <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
Abort message: '/src/platform/android/src/style/sources/custom_geometry_source.cpp:77: void mbgl::android::CustomGeometrySource::cancelTile(const mbgl::CanonicalTileID &): assertion "javaPeer" failed'
    r0 00000000  r1 00002733  r2 00000006  r3 00000008
    r4 00002674  r5 00002733  r6 c7381e38  r7 0000010c
    r8 00002674  r9 d8afec20  sl c8b3f9d5  fp 0000000b
    ip 00000000  sp c7381e28  lr e68b3537  pc e68e3c0c  cpsr 200d0010
backtrace:
    #00 pc 0004ac0c  /system/lib/libc.so (tgkill+12)
    #01 pc 0001a533  /system/lib/libc.so (abort+54)
    #02 pc 0001eb05  /system/lib/libc.so (__libc_fatal+24)
    #03 pc 0001a749  /system/lib/libc.so (__assert2+16)
    #04 pc 0015a18b  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #05 pc 0015c04d  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #06 pc 0015beb9  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #07 pc 0028c575  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #08 pc 00586ca5  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #09 pc 00587125  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #10 pc 00165c77  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #11 pc 00165c19  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #12 pc 0022e8dd  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #13 pc 0022eb2b  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #14 pc 00429eb9  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #15 pc 00429a89  /data/app/com.mapbox.mapboxsdk.testapp-bK3dYHnC_EUYoZyAipKkGA==/lib/arm/libmapbox-gl.so
    #16 pc 00047f2f  /system/lib/libc.so (_ZL15__pthread_startPv+22)
    #17 pc 0001b0ef  /system/lib/libc.so (__start_thread+32)
```

The implicated line:
https://github.com/mapbox/mapbox-gl-native/blob/de59e3a9ce0d8371f584b31f5312cb20d717a96f/platform/android/src/style/sources/custom_geometry_source.cpp#L77

/cc @tobrun @LukasPaczos @jfirebaugh

**Comment by LukasPaczos** [comment]
This is related to https://github.com/mapbox/mapbox-gl-native/issues/12551. I'm going to disable this test case until the fix for #12551 lands.

**Comment by friedbunny** [comment]
Thanks @LukasPaczos — this seems to be intermittent; https://github.com/mapbox/mapbox-gl-native/pull/12462 eventually passed after a couple restarts,
